vwifi.c

```diff
@@ -172,24 +172,34 @@ static void inform_dummy_bss(struct owl_context *owl)
         return;
     hash_for_each_safe (ssid_table, i, tmp, ap, node) {
         struct cfg80211_bss *bss = NULL;
-        struct cfg80211_inform_bss data = {
-            /* the only channel */
-            .chan = &owl->wiphy->bands[NL80211_BAND_2GHZ]->channels[0],
-            .scan_width = NL80211_BSS_CHAN_WIDTH_20,
-            .signal = rand_int(-100, -30) * 100,
-        };
-
-        /* array of tags that retrieved from beacon frame or probe responce */
+        struct ieee80211_channel *channel;
+        s32 signal;
+        u64 tsf;
+        /* IE (information element) that retrieved from
+         * beacon frame or probe responce
+         */
+        struct {
```

Comment:
Don't use anonymous struct unless it is declared with `typedef`.

Reply:
Ok, should I define this structure outside the function?

Reply:
No, you just have to name the structure.